Core/Src/torque_manager.c

```diff
@@ -118,3 +131,18 @@ void send_ref_torque_message (uint32_t *ref_torque) {
 
     osMessageQueuePut(q_ref_torque_messageHandle, &ref_torque_message, 0, 0U);
 }
+
+void control_ramp(uint32_t *ref_torque, uint32_t *ref_torque_ant){
+    for (int i=0;i<2;i++) {
+            if (ref_torque_ant[i] > TORQUE_INIT_LIMITE) {                       // verifica se a referencia
+                if (ref_torque[i] > ref_torque_ant[i] + INC_TORQUE) {           // ja passou do ponto de inflexao
+                    ref_torque[i] = ref_torque_ant[i] + INC_TORQUE;             // e aplica o incremento mais agressivo
+                }
+            } else {
+                if (ref_torque[i] > ref_torque_ant[i] + INC_TORQUE_INIT) {      // caso contrario usa o
+                    ref_torque[i] = ref_torque_ant[i] + INC_TORQUE_INIT;        // incremento da primeira rampa
+                }
+            }
+            ref_torque_ant[i] = ref_torque[i];                                  // aplica a referencia calculada
```

Comment:
Aproveitando q já tá mexendo nisso aqui rsrsrs
```suggestion
            uint32_t torque_increment = (ref_torque_ant[i] > TORQUE_INIT_LIMITE) ? INC_TORQUE : INC_TORQUE_INIT;
            ref_torque[i] = min(ref_torque[i], ref_torque_ant[i] + torque_increment);
```

ai comenta as linhas de acordo com esse novo comportamento